### PR TITLE
AICE-134 basic token usage tracking

### PR DESCRIPTION
**Pull request by bram2000**

### CODE REVIEW

File: app/chat/agent.py [hunk 1]
```diff
@@ -38,11 +38,22 @@ async def execute_flow(
             messages=messages,
         )
 
+        usage = None
+        if response.usage:
+            input_tokens = response.usage.get("input_tokens", 0)
```
Comment (sfitz42) [inline app/chat/agent.py]: I think it would be better to make the default for input and output tokens `None` or throw an error rather than `0`. It's (hopefully) very rare that these are not set as Amazon states they are required in the response.

File: tests/chat/test_bedrock_chat_agent.py
Comment (sfitz42) [inline tests/chat/test_bedrock_chat_agent.py:1]: I think we could do with some more test cases around how the agent handler / repository handle `usage` being set to `None` etc.

File: app/chat/agent.py (hunk 1 above)
Comment (sfitz42) [reply app/chat/agent.py]: Thinking about it again we should just allow the exception to be thrown if the key doesn't exist. If that happens it's either there is something wrong with the response from Bedrock or we've done something wrong internally.

Regardless of the cause, it's something we would prefer to know about rather than lingering in the database.

File: app/chat/repository.py [hunk 2]
```diff
@@ -34,6 +35,9 @@ async def save(self, conversation: models.Conversation) -> None:
                             "role": msg.role,
                             "content": msg.content,
                             "model": msg.model_id,
+                            "usage": dataclasses.asdict(msg.usage)
```
Comment (sfitz42) [inline app/chat/repository.py]: Is there a cleaner way of doing this? A one-liner `if/else` here would be pushing it but putting it onto two lines is confusing.

If the line is too long to satisfy the ruff rules, I think we should create a method to handle this.

**Review by sfitz42 — CHANGES_REQUESTED** [review]
A few changes

File: app/chat/service.py [hunk 3]
```diff
@@ -13,7 +13,7 @@ def __init__(
         self.conversation_repository = conversation_repository
 
     async def execute_chat(
-        self, question: str, model_name: str, conversation_id: uuid.UUID = None
+        self, question: str, model_name: str, conversation_id: uuid.UUID | None = None
     ) -> models.Conversation:
         # Get or create conversation from repository
```
Comment (sfitz42) [inline app/chat/service.py]: Could you remove this comments? I think they're leftovers from a previous AI assistant run

Comment (bram2000) [reply app/chat/service.py]: done

**Review by sfitz42 — APPROVED** [review]
Approved
